src/context/WebsocketContext.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { createContext, useEffect, useState, useContext } from 'react'
+
+import { useCurrentWallet } from './WalletContext'
+
+const WEBSOCKET_RECONNECT_DELAY_STEP = 1_000
+const WEBSOCKET_RECONNECT_MAX_DELAY = 10_000
+
+// webservers will close a websocket connection on inactivity (e.g nginx default is 60s)
+// specify the time in milliseconds at least one 'keepalive' message is sent
+const WEBSOCKET_KEEPALIVE_MESSAGE_INTERVAL = 30_000
+
+// return delay in milliseconds to attempt reconnecting after the connection has been lost
+const connectionRetryDelayLinear = (attempt = 0) => {
+  // linear increase per attempt by `step` amount till `max` is reached
+  const delay = Math.max(WEBSOCKET_RECONNECT_DELAY_STEP, WEBSOCKET_RECONNECT_DELAY_STEP * attempt)
+  return Math.min(delay, WEBSOCKET_RECONNECT_MAX_DELAY)
+}
+
+// path that will be proxied to the backend server
+const WEBSOCKET_ENDPOINT_PATH = '/jmws'
+
+// possible values for property `coinjoin_state` in websocket messages
+const CJ_STATE_TAKER_RUNNING = 0
+const CJ_STATE_MAKER_RUNNING = 1
+const CJ_STATE_NONE_RUNNING = 2
+
+const createWebSocket = () => {
+  const { protocol, host } = window.location
+  const scheme = protocol === 'https:' ? 'wss' : 'ws'
+  const websocket = new WebSocket(`${scheme}://${host}${WEBSOCKET_ENDPOINT_PATH}`)
+
+  if (process.env.NODE_ENV !== 'production') {
+    websocket.onopen = () => {
+      console.debug('websocket connection openend')
+    }
+
+    websocket.onclose = () => {
+      console.debug('websocket connection closed')
+    }
+
+    websocket.onmessage = (event) => {
+      const data = JSON.parse(event?.data)
+      console.debug('websocket message', data)
+    }
+  }
```

Comment:
I'd be in favor of not dumping each message in the console (as it's really a lot of messages). One can also view the messages in the Network tap of browser dev tools (at least in Firefox, probably also in other browsers). That would keep the console clear.